### DISCUSSION TITLE
Flip IN and OUT object counting

### DISCUSSION THREAD

**Discussion by CodeWithEhtisham** [roboflow/supervision#1613] [Q&A] [answered]
I want to count people who are IN or OUT in the stream. I'm using Linezone for this, and it generally works fine. However, there's an issue: when someone enters the house, it incorrectly updates the OUT value, and when someone exits, it updates the IN value. How can I flip these values to make the counts accurate?

**Comment by JShengP** [comment, marked as answer]
Try swapping the start and end points of the LineZone.

**Reply by HamxAnwar** [reply]
This might work but isn't it an improper solution as most of the programming structures follow image line coordinates that run from left to right.

**Reply by LinasKo** [reply]
The answer by @JShengP is correct and here's why:

We typically see a line as a set of 2 points, without a direction. By introducing the "IN" and "OUT" sides, we're adding extra information. It can be encoded in the world space "left side of the image is the IN side of all lines", or on each line: "line goes from A to B, and left side is always IN".

We opt to encode this info on the line - our line zones are therefore implied to always have a direction.

Note that we may add a special case flag to say "this line is reversed", but it's slightly cumbersome. Note that you can do this already by renaming "out:" with "in:" and vice-versa.
